app/src/main/kotlin/org/fossify/filemanager/activities/ReadTextActivity.kt

```diff
@@ -229,7 +230,9 @@ class ReadTextActivity : SimpleActivity() {
                 }
             }
 
-            webView.loadData(binding.readTextView.text.toString(), "text/plain", "UTF-8")
+            val text = binding.readTextView.text.toString()
+            val base64 = Base64.encodeToString(text.toByteArray(), Base64.DEFAULT)
+            webView.loadData(base64, "text/plain", "base64")
```

Comment:
Let's do the encoding in the background to avoid ANRs on large files. You can use the existing `ensureBackgroundThread` helper.